### ISSUE TITLE
Consider improving affordances and focus indicator for column customization sliders

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#4768]
Reported by @kristastarr in the [Datagrid review](https://github.ibm.com/CDAI-design/pal/issues/1575) from the accessibility audit.

> Check that there is a highly visible focus indicator for the column customization sliders

> I feel like it would be helpful for discoverability and comprehension purposes to have a visual indicator present at all times to indicate that there is a slider input to interact with, rather than just changing the cursor on hover. That being said, I know this current implementation matches conventions typically seen elsewhere, so maybe there are a11y guidelines specifically for this interaction that I'm not familiar with.
> 
> This was flagged by the automated checker since **there is no border on the element**. WCAG Guidelines specify a 2px focus indicator with 3:1 contrast https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html, so if the vertical line is 2px wide, technically it 2px greater than the non-focused state (which is not visible). The dot helps. But if we do make a change to make the slider visible at all times, we will need to re-think the focus state in relation to that.

**Comment by elycheea** [comment]
> Reported by @kristastarr in the [Datagrid review](https://github.ibm.com/CDAI-design/pal/issues/1575) from the accessibility audit.
> 
> I feel like it would be helpful for discoverability and comprehension purposes to have a visual indicator present at all times to indicate that there is a slider input to interact with, rather than just changing the cursor on hover. That being said, I know this current implementation matches conventions typically seen elsewhere, so maybe there are a11y guidelines specifically for this interaction that I'm not familiar with.

This could be tricky especially given that data tables are already content-dense. Much of the design discussions that preceded implementation were on reducing cognitive load. If we want to pursue this, may need to open further design discussion with some of the original team. 🤔
